libsel4vmmplatsupport/src/arch/x86/acpi.c

```diff
@@ -257,16 +257,19 @@ int make_guest_acpi_tables(vm_t *vm)
     }
 
     uintptr_t xsdt_addr = lower_bios_addr + (XSDT_START - LOWER_BIOS_START);
+    uintptr_t xsdt_vaddr = XSDT_START;
 
     acpi_xsdt_t *xsdt = calloc(1, xsdt_size);
     acpi_fill_table_head(&xsdt->header, "XSDT", 1);
 
     // Add previous tables to XSDT pointer list
     uintptr_t table_paddr = xsdt_addr + xsdt_size;
+    uintptr_t table_vaddr = xsdt_vaddr + xsdt_size;
     uint64_t *entry = (uint64_t *)((char *)xsdt + sizeof(acpi_xsdt_t));
     for (int i = 1; i < num_tables; i++) {
-        *entry++ = (uint64_t)table_paddr;
+        *entry++ = (uint64_t)table_vaddr;
         table_paddr += table_sizes[i];
```

Comment:
Seems the value of `table_paddr` is no longer used here? So the initialization and update could be removed for this loop.